File: problem-solving/_lc-387-easy-first-unique-character-in-a-string/1-solution.py
"""
https://leetcode.com/problems/first-unique-character-in-a-string/

Complexity
----------
Time: O(n)
Space: O(n)
"""

class Solution:
    def firstUniqChar(self, s: str) -> int:
        count = {}
        for c in s:
            if c not in count:
                count[c] = 0
            count[c] += 1

        for i in range(len(s)):
            if count[s[i]] == 1:
                return i
        return -1

File: problem-solving/_lc-27-easy-remove-element/1-solution.go
/*
https://leetcode.com/problems/remove-element/

Complexity
----------
Time: O(n)
Space: O(1)
*/

func removeElement(nums []int, val int) int {
    var i int
    for _, x := range nums {
        if x != val {
            nums[i] = x
            i += 1
        }
    }
    return i
}
